Casks/s/sapmachine-jdk.rb

```diff
@@ -15,7 +15,19 @@
   # following JSON file, so we have to check it instead.
   livecheck do
     url "https://sap.github.io/SapMachine/assets/data/sapmachine-releases-latest.json"
-    regex(/["']tag["']:\s*["']sapmachine[._-]v?(\d+(?:\.\d+)*)["']/i)
+    regex(/^\D*(\d+(?:\.\d+)+)$/i)
```

Comment:
```suggestion
    regex(/^sapmachine[._-]v?(\d+(?:\.\d+)*)$/i)
```

We have to use the optional `*` form of the standard regex, as major release tags use a format like [`sapmachine-23`](https://github.com/SAP/SapMachine/releases/tag/sapmachine-23), similar to Java releases.

There may also be something to be said for using a `sapmachine[._-]` prefix, as the upstream repository also contains older tags with a `jdk-` prefix. For what it's worth, my previous comment about `\D*` was more of a situational suggestion rather than common guidance, as the looseness of `\D*` can potentially lead to unwanted matches (depending on the context).

Reply:
That makes sense, thank you!